### ISSUE TITLE
Add the option to export s5cmd compatible manifest

### ISSUE THREAD

**Issue by fedorov** [ImagingDataCommons/IDC-WebApp#1155]
s5cmd manifest is a list of commands that are executed by the s5cmd command (as described in https://learn.canceridc.dev/data/downloading-data/downloading-data-with-s5cmd).

Here's example query that will make manifests that will be compatible with the series-level reorganized buckets:

```sql
WITH
  tempt AS (
  SELECT
    STRING_AGG(DISTINCT(SPLIT(gcs_url,'/')[SAFE_OFFSET(2)])) AS bucket_name,
    crdc_series_uuid
  FROM
    bigquery-public-data.idc_current.dicom_all
  GROUP BY
    crdc_series_uuid)
SELECT
  CONCAT("cp s3://",bucket_name,"/",crdc_series_uuid) AS series_cp_command
FROM
  tempt
```

**Comment by fedorov** [comment]
Per discussion today:
* agreed to add the option to export s5cmd manifest in such a way that it can be passed directly to s5cmd with no edits by the user
* will add tabs to handle export of different types of manifest (files, s5cmd, BQ), with s5cmd probably selected by default
* will test how big the cohort should be to exceed the allowed size for the exported manifest

In the future, might consider zipping manifest on download.

**Comment by pgundluru** [comment]
This ticket has been tested in windows command line and MacOS (Linux) 

1. In windows using the command line terminal - Users will need to install google SDK on command line to be able to follow the steps in the doc https://learn.canceridc.dev/data/downloading-data/downloading-data-with-s5cmd there is no mention of installing google SDK so that might be confusing to some users. 
2. The first box could be specified as an example of what the manifest should contain. The second box doesn't copy the 2 lines when copied to terminal so maybe we could remove the line break 
![image](https://user-images.githubusercontent.com/85526819/234902607-3b70a89c-e595-47ab-8240-a66d7a48805e.png)

**Comment by pgundluru** [comment]
Tested this by creating a cohort and can see the tabbed interfaces for export manifest. s5cmd is not selected on default since the CSV/TSV/JSON tab is default first. 

I have created a cohort with all collections and body parts and primary site (over 47k series collections) and was able to export manifest to s5cmd via AWS and GCP

**Comment by pgundluru** [comment]
![image](https://user-images.githubusercontent.com/85526819/234905013-2c89eb71-8a78-466b-93d3-ab43b2d1a9fd.png)

**Comment by fedorov** [comment]
> In windows using the command line terminal - Users will need to install google SDK on command line to be able to follow the steps in the doc https://learn.canceridc.dev/data/downloading-data/downloading-data-with-s5cmd there is no mention of installing google SDK so that might be confusing to some users.

I did notice this earlier today, and it is already fixed!

![image](https://user-images.githubusercontent.com/313942/234925308-539d9d61-7752-44a7-a7d0-90ead519921e.png)

> The second box doesn't copy the 2 lines when copied to terminal so maybe we could remove the line break

Done

> s5cmd is not selected on default since the CSV/TSV/JSON tab is default first.

Nice catch! I think this is not good. The TSV/CSV manifest is pretty much useless. @s-paquette I suggest we move s5cmd manifest to be the default.

**Comment by s-paquette** [comment]
Can do--I'll also shift it left. Note though that since s5cmd and CSV/etc. is only for current versions, they will always be disabled on older version cohorts, making BQ the default.

**Comment by fedorov** [comment]
Few more items to update:
* manifest file name should include aws or gcs, depending on the location of the buckets
* need to check if we can add comment at the top of the manifest
* GCP option should be first on the list and should be selected by default

**Comment by fedorov** [comment]
I think we should also have a link to the download instructions - users most likely will not know what s5cmd is.

**Comment by s-paquette** [comment]
@fedorov 'need to check if we can add comment at the top of the manifest' Comment as in, commented out fields?

**Comment by s-paquette** [comment]
@pgundluru @fedorov All items except 'need to check if we can add comments' are now on test. Note that the aws/gcs part of the filename won't show in the box because it is appended on the backend, to prevent it being accidentally removed by the user (similar to file part names under the CSV download).

**Comment by fedorov** [comment]
* the manifest does not work - as also discovered by @G-White-ISB in https://nciimagingdat-m8a6349.slack.com/archives/C045VQUAZ0S/p1682646788735799?thread_ts=1682643952.876529&cid=C045VQUAZ0S, we need to add wildcard to the command, i.e., `cp s3://idc-open-data/6f146b7a-9060-4e20-9344-9d29dcbae798/* .` instead of `cp s3://idc-open-data/6f146b7a-9060-4e20-9344-9d29dcbae798/ .`
* `s5cmd` support comments via `#` in the manifest; we can add the same header as we do for CSV/TSV, plus we can include in the comment the command line that needs to be executed to download the content of the manifest (picking the appropriate s5cmd command between GCP and AWS, depending on the user selection):
> `# To download the files in this manifest, first install s5cmd (https://github.com/peak/s5cmd), `
> `# then run the following command:`
> `# s5cmd --no-sign-request --endpoint-url https://s3.amazonaws.com run <manifest name>`
> `# s5cmd --no-sign-request --endpoint-url https://storage.googleapis.com <manifest name>`
* the dialog does not look right - I think s5cmd tab should be highlighted: 
![image](https://user-images.githubusercontent.com/313942/235167210-7a766c76-5213-4841-a442-464bad6c775a.png)
* let's add the line "Please see instructions on how to download files defined by the manifest using `s5cmd` here: https://learn.canceridc.dev/data/downloading-data" below the radio buttons.

**Comment by G-White-ISB** [comment]
Actually if you want to download  the files in the hierarchy then the correct format would be  cp s3://idc-open-data/6f146b7a-9060-4e20-9344-9d29dcbae798/ ./6f146b7a-9060-4e20-9344-9d29dcbae798 

But here's confusing thing:
If I try 
s5cmd --no-sign-request cp s3://idc-open-data/6f146b7a-9060-4e20-9344-9d29dcbae798/* ./6f146b7a-9060-4e20-9344-9d29dcbae798

I get an error message,
 no matches found: s3://idc-open-data/6f146b7a-9060-4e20-9344-9d29dcbae798/*

However if I write the line
 cp s3://idc-open-data/6f146b7a-9060-4e20-9344-9d29dcbae798/* ./6f146b7a-9060-4e20-9344-9d29dcbae798
to a file named cohort.txt and then try

s5cmd --no-sign-request run cohort.txt

the download works.

**Comment by fedorov** [comment]
Problem is, if we want to keep the hierarchy, we will have a much larger manifest file for the same number of entries, which will limit the maximum size of the manifest. I think it might be better to download into a flat space.

**Comment by fedorov** [comment]
> But here's confusing thing:
> If I try
> s5cmd --no-sign-request cp s3://idc-open-data/6f146b7a-9060-4e20-9344-9d29dcbae798/* ./6f146b7a-9060-4e20-9344-9d29dcbae798
>
>I get an error message,
>no matches found: s3://idc-open-data/6f146b7a-9060-4e20-9344-9d29dcbae798/*

I think you just need to put the part that has * in quotes.

**Comment by s-paquette** [comment]
@G-White-ISB @fedorov Do I need to adjust how the command is produced? (Working on adding in the header lines.)

**Comment by G-White-ISB** [comment]
@s-paquette you just replace s3://bucket/series_uuid/ . with s3://bucket/series_uuid/* .

**Comment by fedorov** [comment]
> Do I need to adjust how the command is produced? (Working on adding in the header lines.)

@s-paquette I answered that question in bullet 1 of https://github.com/ImagingDataCommons/IDC-WebApp/issues/1155#issuecomment-1527643105.

**Comment by s-paquette** [comment]
@fedorov @G-White-ISB I was referring to the comment about the quotes from this:

> I think you just need to put the part that has * in quotes.

Should we be putting quotes in the manifest as well, just in case, or will that break use of s5cmd?

**Comment by fedorov** [comment]
No, when I tested on my laptop, I did not need to add any quotes when the commands are specified in the manifest file.

**Comment by pgundluru** [comment]
Bug that showed SV/TSV/JSON instead of S5cmd i fixed 
![image](https://user-images.githubusercontent.com/85526819/235499590-addefc0d-ecb6-49de-a6af-3ed2483b93c0.png)

**Comment by pgundluru** [comment]
Able to see the comments in cohort manifest using s5cmd 
![image](https://user-images.githubusercontent.com/85526819/235499841-4dfea88f-1f8a-417e-86fa-4faac06f4445.png)

**Comment by s-paquette** [comment]
@fedorov I was able to get the link to work without closing the dialog. Did we possibly want to reword this to something simpler, such as "Download Instructions" and just have a link? Or "Manifest instructions", something simpler?

**Comment by fedorov** [comment]
Nice!

How about "[Instructions](https://learn.canceridc.dev/data/downloading-data) on downloading images defined by the manifest"?

**Comment by fedorov** [comment]
Let's update the button text to read "Download s5cmd manifest" in place of "Download S5cmd" in the below:

<img width="475" alt="image" src="https://user-images.githubusercontent.com/313942/236028989-ba020f6c-2875-4bcf-8845-e61a2b88f51b.png">

**Comment by fedorov** [comment]
Yes!
